docs/tutorials/kubernetes-hpa.md

```diff
@@ -0,0 +1,313 @@
+* `nodeinfo` is the name of the function
+* `--cpu-percentage` is the level of CPU the Pod should reach before additional replicas are added
+* `--min` minimum number of Pods
+* `--max` maximum number of Pods
+
+View the HPA record with:
+
+```
+kubectl get hpa/nodeinfo -n openfaas-fn
+
+NAME       REFERENCE             TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
+nodeinfo   Deployment/nodeinfo   <unknown>/50%   1         10        0          15s
+
+# view the YAML
+
+kubectl get hpa/nodeinfo -n openfaas-fn -o yaml
+```
+
+You can use `kubectl describe hpa/nodeinfo -n openfaas-fn` to get detailed information including any events such as scaling up and down.
+
+```
+kubectl describe hpa/nodeinfo -n openfaas-fn
+
+Name:                                                  nodeinfo
+Namespace:                                             openfaas-fn
+Labels:                                                <none>
+Annotations:                                           <none>
+CreationTimestamp:                                     Sat, 10 Aug 2019 11:48:42 +0000
+Reference:                                             Deployment/nodeinfo
+Metrics:                                               ( current / target )
+  resource cpu on pods  (as a percentage of request):  20% (2m) / 50%
+Min replicas:                                          1
+Max replicas:                                          10
+Deployment pods:                                       4 current / 4 desired
+Conditions:
+  Type            Status  Reason               Message
+  ----            ------  ------               -------
+  AbleToScale     True    ScaleDownStabilized  recent recommendations were higher than current one, applying the highest recent recommendation
+  ScalingActive   True    ValidMetricFound     the HPA was able to successfully calculate a replica count from cpu resource utilization (percentage of request)
+  ScalingLimited  False   DesiredWithinRange   the desired count is within the acceptable range
+Events:           <none>
+```
+
+### Generate some traffic
+
+Use the following command with hey
+
+```
+export OPENFAAS_URL=http://127.0.0.1:31112
+
+hey -c 5 \
+ -z 5m \
+ $OPENFAAS_URL/function/nodeinfo
+```
+
+* `-c` will simulate `5` concurrent users
+* `-z` will run for `5m` and then complete
+
+You should note that HPA is designed to react slowly to changes in traffic, both for scaling up and for scaling down. In some instances you may wait 20 minutes for all your Pods to scale back down to a normal level after the load has stopped.
+
+Now in a new window monitor the progress:
+
+```
+kubectl describe hpa/nodeinfo -n openfaas
+```
+
+Or in an automated fashion:
+
+```
+# watch -n 5 "kubectl describe hpa/nodeinfo -n openfaas-fn"
+```
+
+You can also monitor the replicas of your function in the OpenFaaS UI or via the CLI:
+
+```
+watch -n 5 "faas-cli list"
+```
+
+Here is ane example of the replicas scaling up in response to the traffic created by `hey`:
```

Comment:
ane->an

Reply:
Good catch. Would you mind trying this out?